src/main/java/net/dries007/tfc/util/Helpers.java

```diff
@@ -781,6 +781,10 @@ public static boolean detectAssertionsEnabled()
      */
     public static boolean detectTestSourcesPresent()
```

Comment:
This needs to be two seperate methods, as the name does not imply what it is doing

- `Helpers.detectBootstrapEnvironment()` - Both checks. This is an environment where we aren't launched through a normal server, and so stuff like the hack in `TFCLayers` needs to do something different, to avoid crashing. That's the current environment in which `/gradlew test` is ran through.
- `Helpers.detectTestSourcesPresent()` - Just if test sources are present, that's all.